include/natalie/io_object.hpp

```diff
@@ -95,6 +96,7 @@ class IoObject : public Object {
     void set_fileno(int fileno) { m_fileno = fileno; }
     Value set_lineno(Env *, Value);
     Value set_sync(Env *, Value);
+    void set_nonblock(Env *, bool) const;
```

Comment:
I had to make this one `const` to make clang-tidy pass, but I can't say I agree with this analysis. Yes, from a compilers point of view we don't mutate the object, but it's still a mutation of something inside the object.